ui/components/app/edit-gas-popover/edit-gas-popover.component.js

```diff
@@ -55,26 +94,42 @@ export default function EditGasPopover({
     if (!transaction || !mode) {
       closePopover();
     }
+
+    const cancelSpeedUpGas = process.env.SHOW_EIP_1559_UI
+      ? {
+          gasLimit: decimalToHex(gasLimit),
```

Comment:
I need to figure out the root cause of `Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: 0x5.100000001` when using this `decimalToHex` function.

Reply:
Ah, you need to convert these using `decGWEIToHexWEI` from `shared/modules/conversion.util.js` it does two conversions, first from gwei to wei then dec to hex. Not gasLimit though, gasLimit is just dec to hex string.